### ISSUE TITLE
ConnectionError: Login Failed for user <username>

### ISSUE THREAD

**Issue by Jeordie** [mtxr/vscode-sqltools#67]
> This is a guideline to help us to understand SQLTools Issues. 
> 
> You don't need to stick to this, but please try to guide us to reproduce the errors or understand your feature requests.

### Issue Type

* [ ] Feature Request
* [x] Bugfix/Errors
* [ ] Questions
* [ ] Other

### Prerequisites (For bugfixes)

* [x] Are you running the latest version?
* [x] Did you check the logs?
* [x] Did you check the Setup?

### Description

As stated in issue nr 65. I'm still not able to connect to sql server. Resolving Issue #60 didn't bring a solution to my problem. I'm using the following information to connect:

"sqltools.connections": [
        {
            "name": "Test",
            "server": "svr-t-evasql01.otapp.local",
            "dialect": "MSSQL",
            "port": 1433,
            "database": "EvaSfh",
            "username": "domainname\Jeordiep",
            "askForPassword": false,
            "connectionTimeout": 60
        }
    ]
I use integrated security = true to connect to sql server

The following error occurs in the log:

Query error: { ConnectionError: Login failed for user 'domainname\Jeordiep\'.
    at ConnectionError (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\errors.js:12:12)
    at Parser.<anonymous> (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\connection.js:507:33)
    at emitOne (events.js:96:13)
    at Parser.emit (events.js:191:7)
    at Parser.<anonymous> (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\token\token-stream-parser.js:54:15)
    at emitOne (events.js:96:13)
    at Parser.emit (events.js:191:7)
    at addChunk (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:284:12)
    at readableAddChunk (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:271:11)
    at Parser.Readable.push (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:238:10)
  message: 'Login failed for user \'domainname\Jeordiep\'.',
  code: 'ELOGIN' }
Error while preparing columns completions { ConnectionError: Login failed for user '<domainname>\\Jeordiep'.
    at ConnectionError (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\errors.js:12:12)
    at Parser.<anonymous> (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\connection.js:507:33)
    at emitOne (events.js:96:13)
    at Parser.emit (events.js:191:7)
    at Parser.<anonymous> (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\tedious\lib\token\token-stream-parser.js:54:15)
    at emitOne (events.js:96:13)
    at Parser.emit (events.js:191:7)
    at addChunk (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:284:12)
    at readableAddChunk (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:271:11)
    at Parser.Readable.push (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.0\node_modules\readable-stream\lib\_stream_readable.js:238:10)
  message: 'Login failed for user \'domainname\Jeordiep\'.',
  code: 'ELOGIN' }

**Comment by Jeordie** [comment]
I have to use a domain name when logging in so: <domain>\jeordiep. I also tried it with double \\

**Comment by mtxr** [comment]
You are right. i forgot to consider de domain parameter in the options as de TDS driver needs (http://tediousjs.github.io/tedious/api-connection.html#function_newConnection).

I'll release a fix for it later today, ok? Then you will be able to connect using this setup:

```json
{
  "name": "Test",
  "server": "svr-t-evasql01.otapp.local",
  "dialect": "MSSQL",
  "port": 1433,
  "database": "EvaSfh",
  "domain": "domainname",
  "username": "Jeordiep",
  "askForPassword": false,
  "connectionTimeout": 60
}
```

if you are using an azure server, you probably need to include the encrypt flag as following:

```json
{
  "name": "Test",
  "server": "svr-t-evasql01.otapp.local",
  "dialect": "MSSQL",
  "port": 1433,
  "database": "EvaSfh",
  "domain": "domainname",
  "username": "Jeordiep",
  "askForPassword": false,
  "connectionTimeout": 60,
  "dialectOptions": {
    "encrypt": true
  }
}
```

**Comment by mtxr** [comment]
Closed on #68.

**Comment by Jeordie** [comment]
Hi I'm still not able to connect: I'm setting up a connection using the connection wizard (great add on). After that I'm trying to connect. The log shows me this error:

[2018-03-05T07:49:34.][v0.12.1] ERROR: Error selecting connection: 
[2018-03-05T07:49:34.][v0.12.1] Error: Connection got disposed.
	at Object.dispose (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\main.js:825:25)
	at Object.dispose (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-languageclient\lib\client.js:57:35)
	at LanguageClient.handleConnectionClosed (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-languageclient\lib\client.js:1864:38)
	at LanguageClient.handleConnectionClosed (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-languageclient\lib\main.js:106:15)
	at closeHandler (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-languageclient\lib\client.js:1852:18)
	at CallbackList.invoke (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\events.js:71:39)
	at Emitter.fire (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\events.js:135:36)
	at closeHandler (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\main.js:221:26)
	at CallbackList.invoke (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\events.js:71:39)
	at Emitter.fire (C:\Users\jeordiep\.vscode\extensions\mtxr.sqltools-0.12.1\node_modules\vscode-jsonrpc\lib\events.js:135:36)
